### ISSUE TITLE
Electron-builder ignoring package.json build tag.

### ISSUE THREAD

**Issue by davimello28** [electron-userland/electron-builder#4180]
* **Version**: 21.2.0
* **Target**: win, portable
* **OS**: Linux Debian 10 (I think this is important). 

I was unable to use the `"build"` tag on my package.json. It is important to mention that `electron-builder -w portable` works as expected. 

Here is my `package.json`:
```json
{
 ...
  "scripts": {
    "pack": "electron-builder --dir",
    "dist": "electron-builder",
    "postinstall": "electron-builder install-app-deps"
  },
  "build": {
    "appId": "nonames.sad",
    "productName": "noluck",
    "win": {
      "target": [
        "portable"
      ]
    }
  }
}
```
...which wrongly produces:
```
  • packaging       platform=linux arch=x64 electron=6.0.2 appOutDir=dist/linux-unpacked
  • building        target=snap arch=x64 file=dist/emissor-boletos_1.0.0_amd64.snap
  • building        target=AppImage arch=x64 file=dist/noluck-1.0.0.AppImage
```

I've tried to search for this issues and found some others having similar problems. 

#4075, #3612, #1410

#3543 - I've tested with some other builds included (Ex: `nsis`). It does not work also. 

Curiously, the `builder-effective-config.yaml` seems to be interpreted correctly:
```yml
directories:
  output: dist
  buildResources: build
appId: nonames.sad
productName: noluck
win:
  target: portable
electronVersion: 6.0.2
```
and the DEBUG shows me exactly the line that is passing the wrong argument:

```
• spawning        command=.../node_modules/app-builder-bin/linux/x64/app-builder unpack-electron --configuration [{"platform":"linux","arch":"x64","version":"6.0.2"}] --output .../dist/linux-unpacked --distMacOsAppName Electron.app
```

Here is the full debug:
```
  • electron-builder  version=21.2.0 os=4.19.0-5-amd64
  • loaded configuration  file=package.json ("build" field)
  • effective config  config=
                        directories:
                          output: dist
                          buildResources: build
                        appId: nonames.sad
                        productName: noluck
                        win:
                          target: portable
                        
  • writing effective config  file=dist/builder-effective-config.yaml
  • spawning        command=[...]/node_modules/app-builder-bin/linux/x64/app-builder node-dep-tree --dir [...]
  • exited          command=app-builder code=0 pid=4527 out=[]
  • spawning        command=[...]/node_modules/app-builder-bin/linux/x64/app-builder rebuild-node-modules cwd=[...]
  • no native dependencies  
  • exited          command=app-builder code=0 pid=4532
  • packaging       platform=linux arch=x64 electron=6.0.2 appOutDir=dist/linux-unpacked
  • spawning        command=[...]/node_modules/app-builder-bin/linux/x64/app-builder unpack-electron --configuration [{"platform":"linux","arch":"x64","version":"6.0.2"}] --output [...]/dist/linux-unpacked --distMacOsAppName Electron.app
```

## Things I've tested

I spent my day full on this issue:

- Providing a YML file instead of JSON.
- Inputing more than one tag.

**Comment by davimello28** [comment]
Another test that I did was extracting the `builder-effective-config.yaml` from `/tests` inside the repo and use it to run as my `electron-builder.yml`.

Here is the `builder-effective-config.yaml` exported from:
```bash
TEST_FILES="portableTest"
node ./test/out/helpers/runTests.js
```

```yaml
directories:
  output: dist
  buildResources: build
electronVersion: 6.0.0
appId: org.electron-builder.testApp
compression: store
npmRebuild: false
mac:
  category: your.app.category.type
  identity: null
linux:
  category: Development
deb:
  packageCategory: devel
squirrelWindows:
  iconUrl: 'https://raw.githubusercontent.com/szwacz/electron-boilerplate/master/resources/windows/icon.ico'
nsis:
  artifactName: '${productName}Installer.${version}.${ext}'
  installerIcon: foo test space.ico
portable:
  requestExecutionLevel: admin
  artifactName: '${productName}Portable.${version}.${ext}'
```
 
And again...this was the result:
```
  • electron-builder  version=21.2.0 os=4.19.0-5-amd64
  • loaded configuration  file=[...]/electron-builder.yml
  • writing effective config  file=dist/builder-effective-config.yaml
  • skipped dependencies rebuild  reason=npmRebuild is set to false
  • packaging       platform=linux arch=x64 electron=6.0.0 appOutDir=dist/linux-unpacked
  • building        target=snap arch=x64 file=dist/emissor-boletos_1.0.0_amd64.snap
  • building        target=AppImage arch=x64 file=dist/emissor-boletos-1.0.0.AppImage

```

**Comment by davimello28** [comment]
I think that I found the answer.

After some research, I found out that the `electron-builder` does not consider the configuration file (in my case `package.json`, but it can be the `electron-builder.yml` if you specify the `--config`) file by itself. It **needs** the target tags (Eg: `-w`, `-l` and `-m` for Windows, Linux and macOS) to build the correct packages. 

When it doesn't receive the target , it automatically builds it to **the current platform** (in my case, it is Linux). I end up creating some NPM scripts to build for different platforms and used the default `npm run dist` to publish to all the platforms.

```json
{
 "dist": "electron-builder -mlw portable --config",
 "dist-linux": "electron-builder -l --config",
 "dist-win": "electron-builder -w portable --config",
 "dist-mac": "electron-builder -m --config"
}
```
It is important to mention that the documentation is missleading. The Quick Guide points the following:

> Then you can run yarn dist (to package in a distributable format (e.g. dmg, windows installer, deb package)) or yarn pack (only generates the package directory without really packaging it. This is useful for testing purposes).

Since the `yarn dist` in the exemple includes the configuration file (`package.json`) previously, you are misleaded to believe that it will build for the configured platform. **This does not build for the configured platform specified in the configuration file. This builds to your current system, even if you do not specify anything about your current system in the configuration file.**

I am creating a pull request for the documentation to correct that and also to correct #4075.
